Milestone#3

scottlittle18 opened the pull request:
Milestone #3 objectives complete. Just as a note though, the TorchTriggerVolume script isn't complete and is still being worked on which is why it's pretty empty.

scottlittle18 committed:
Line up stairs and add stutue text standup

scottlittle18 committed:
Add Player Interactivity, Reticle & Make TextMeshPro Object for the BrokenCar

- Added new script to handle player interactivity
- Added a '+' with an outline as the reticle on a canvas
- Created new TMPro asset that appears when the player interacts with the BrokenCar object

scottlittle18 committed:
Remove Unnecessary Debug Statements Related to BrokenCar TMPro Object

No Major changes made other than the removal of debug code that was no longer needed

scottlittle18 committed:
Change Assets to Fit the new premise

- Removed the smoke particle effect from the Car Prototype asset
- Changed the name of
 the BrokenCar_TextMesh to Car_TextMesh

scottlittle18 committed:
Add Doors to the other buildings & Animate the unlocked Temple key storage structure

- Added doors to all of the buildings
- Animated the first structure's door
- Added a text mesh to the locked temple key storage building
- TextMesh for locked building needs to be programmed to display like the car text

scottlittle18 committed:
Create new Interface script for doors and Fix Title Menu

- Title Menu title panel works correctly now
- Door Interface script is made but not complete or implamented

scottlittle18 committed:
Add MainMenu Script to TitleMenu scene's Camera

- Added a Credit Layout that appears when the Credits button is clicked
- Shrunk Main Menu Panel to be more focused in the center of the screen

scottlittle18 committed:
Add Main Menu Camera Animation

- Added the animation to the main menu camera
- Removed the Credits method from the Main Menu Script

scottlittle18 committed:
Add Place holders for the Teleportation Objects

scottlittle18 committed:
Animate Doors and Create KeyStorage Locations

- Created Key storage locations that the player will be teleported to
- Animated all doors, but did not program them accordingly since I will be working on Locked Doors soon
- Added Transport targets to each KeyStorage Locations, did not add return points yet

scottlittle18 committed:
Teleporter Script Works

- Teleporter script works, but the parent class will be fixed later, I'm just still learning how to work with inheretance in Unity

scottlittle18 committed:
Add TeleportationObject Script to TempleKey stand in

scottlittle18 committed:
Add TextMeshObjects tag and coded the TextMoshPro Texts to disappear of start

- Updated scripts to make the text mesh for each door display for a set amount of time

scottlittle18 committed:
Fix Teleporter Destruction Delay

- Fixed the bug that kept the teleporter objects from being destroyed

scottlittle18 committed:
Start scripting the Torches to get turned on when the player enters the trigger volume

DirtyDanChicago commented on Assets/My Assets/Scripts/Door_Unlocked.cs:
```diff
@@ -0,0 +1,19 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
```
Add a description.

DirtyDanChicago commented on Assets/My Assets/Scripts/IDoor.cs:
```diff
@@ -0,0 +1,13 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
```
Add a description.

DirtyDanChicago commented on Assets/My Assets/Scripts/MainMenu.cs:
```diff
@@ -0,0 +1,26 @@
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.SceneManagement;
+
```
Add a description.

DirtyDanChicago commented on Assets/My Assets/Scripts/InteractionBasedTeleporter.cs:
```diff
@@ -0,0 +1,24 @@
+        teleportationTarget = target;
+    }
+
+    //public virtual void TeleportPlayer()
```
Not an issue, but if you do /* at the top, and */ at the bottom of the area you want to be commented out it saves you a lot of trouble.

DirtyDanChicago commented on Assets/My Assets/Scripts/TeleporterObject.cs:
```diff
@@ -0,0 +1,25 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
```
Add a description.

DirtyDanChicago commented on Assets/My Assets/Scripts/TorchTriggerVolume.cs:
```diff
@@ -0,0 +1,23 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
```
Add description.

DirtyDanChicago reviewed (changes requested):
Pretty much just add more summaries to the beginning of your scripts.

scottlittle18 committed:
Complete Requested Changes to Scripts

- Added descriptions to all scripts
- Removed the TeleportPlayer() from the InteractionBasedTeleporter script
